test/puppetlabs/trapperkeeper/testutils/logging_test.clj

```diff
@@ -0,0 +1,150 @@
+(ns puppetlabs.trapperkeeper.testutils.logging-test
+  (:require
+   [clojure.test :refer :all]
+   [clojure.tools.logging :as log]
+   [puppetlabs.kitchensink.core :as kitchensink]
+   [puppetlabs.trapperkeeper.logging :refer [reset-logging root-logger-name]]
+   [puppetlabs.trapperkeeper.testutils.logging :as tgt :refer [event->map]])
+  (:import
+   [org.slf4j Logger LoggerFactory]))
+
+;; Without this, "lein test NAMESPACE" and :only invocations may fail.
+(use-fixtures :once (fn [f] (reset-logging) (f)))
+
+(deftest with-log-level-and-logging-to-atom
+  (let [expected {:logger "puppetlabs.trapperkeeper.testutils.logging-test"
+                  :level :info
+                  :message "wlta-test"
+                  :exception nil}]
+    (let [log (atom [])]
+      (tgt/with-log-level root-logger-name :error
+        (tgt/with-logging-to-atom root-logger-name log
+          (log/info "wlta-test"))
+        (is (not-any? #(= expected %) (map event->map @log)))))
+    (let [log (atom [])]
+      (tgt/with-log-level root-logger-name :info
+        (tgt/with-logging-to-atom root-logger-name log
+          (log/info "wlta-test"))
+        (is (some #(= expected %) (map event->map @log)))))))
+
+(def find-logger #'puppetlabs.trapperkeeper.testutils.logging/find-logger)
+(def log-event-listener #'puppetlabs.trapperkeeper.testutils.logging/log-event-listener)
```

Comment:
@cprice404 if "we" might mean just me at the moment, then yes?  It's just a way I've used to test things that are private and/or use them in tests.